### DISCUSSION TITLE
CLIMO_MEAN_HOUR_INTERVAL = "NA" Not Working

### DISCUSSION THREAD

**Discussion by ShannonShields-NOAA** [dtcenter/METplus#1786] [Configuration] [answered]
Hi, I am trying to test a CFS monthly climo file in METplus 4.1.1 on Hera. The MET site says "Use “NA” if the timing of the climatology data should not be checked."
However, when I set GRID_STAT_CLIMO_MEAN_HOUR_INTERVAL = "NA" I receive the following error:
09/08 20:17:46.740 metplus (config_metplus.py:841) ERROR: [config] GRID_STAT_CLIMO_MEAN_HOUR_INTERVAL must be an integer.
If I leave the HOUR_INTERVAL blank, then METplus says:
DEBUG 3: Skipping the 19910831_000000 "t2m(*,*,*)" climatology field since the time offset (64800 seconds) >= the "hour_interval" entry (21600 seconds) from file: /scratch2/NCEPDEV/stmp1/Shannon.Shields/scripts/s2s_data/climo_files/cfs_clim/CFSv2T2mClim19912020_08_cf_fix.nc
I'm not sure why I can't set GRID_STAT_CLIMO_MEAN_HOUR_INTERVAL = "NA" as the documentation says.
I appreciate your assistance; thank you!

**Comment by georgemccabe** [comment]
Hi @ShannonShields-NOAA, this is a bug that was identified and fixed for issue #1653 and is available in the METplus 4.1.3 release. This release is available on Hera, so you should be able to run `module load metplus/4.1.3` instead of `module load metplus/4.1.1` to get the necessary updates.

**Reply by ShannonShields-NOAA** [reply]
Thanks George; I suspected that may be the issue. However, when I use METplus 4.1.3 I am still receiving the error that GRID_STAT_CLIMO_MEAN_HOUR_INTERVAL must be an integer...I have tried setting the hour interval to both "NA" and NA. I tried this with MET 10.1.1 and MET 10.1.2. I am not sure why the bug fix isn't working...

**Reply by georgemccabe** [reply]
It looks like the fix was only applied to day_interval and not hour_interval. Sorry, I had misread the issue. I can create a new bugfix release and install it on Hera. I will reply here when that is available.

**Reply by ShannonShields-NOAA** [reply]
Ah ok, thank you. FYI, this fix is needed to read the CFS monthly climo files into METplus for the seasonal aspect of EVS.

**Comment by georgemccabe** [comment, marked as answer]
Hi @ShannonShields-NOAA, the METplus 4.1.4 bugfix release is now available on Hera.

`module load metplus/4.1.4`

Let me know if you have any other issues.

**Reply by georgemccabe** [reply]
@ShannonShields-NOAA, does this bugfix release need to be installed on any other machines for your work?

**Reply by ShannonShields-NOAA** [reply]
Thank you for getting that fixed so quickly! I am done for the day but will make sure to test this fix first thing Monday and let you know the results. Yes, we will need this for EVS on WCOSS2 (need to install and test on Acorn first).

**Reply by georgemccabe** [reply]
OK, sounds good. Let me know when you have tested on Acorn and @jprestop can install on WCOSS2.

**Reply by jprestop** [reply]
@georgemccabe NCAR employees actually are not allowed to install software for users on WCOSS2  (although we were on WCOSS) for use or for testing changes, updates, modifications, etc. to software.  Some of us have access to install software on acorn, simply to test that the installation works.  It is my understanding that any testing of software needs to be done on other NOAA machines like, Hera, Jet, Gaea, and Orion, which we do have access to install for testing.  Updates to the METplus components software on WCOSS2 needs to be a request to NCO (I do not know of the process but perhaps @AliciaBentley-NOAA can advise).

**Reply by AliciaBentley-NOAA** [reply]
@georgemccabe @jprestop That's correct. @ShannonShields-NOAA can test whether or not the bug fix works on Hera on Monday morning. If it does, then Julie can build and test an installation on Acorn. If that also looks good, then Jason Levit can ask NCO to formally install the bug fix on WCOSS2. 

Time is of the essence with this bug fix (NCO installing anything can take forever and we need this fix long before the EVSv1 alpha test on 12/1/22). We really appreciate your help!

**Reply by jprestop** [reply]
@AliciaBentley-NOAA Because installation of METplus is simply "unpack the METplus software" no test of installation needs to be done on Acorn.  Once @ShannonShields-NOAA give the go ahead on the test of the bugfix, it's ready for Jason to ask NCO to formally install the bugfix.  If it were MET and needed to be recompiled, we would want to test installation on Acorn for sure, but that is not necessary in this case.

**Reply by jprestop** [reply]
From the METplus User's Telecon this week, we know that this functionality was tested by Shannon on Hera and works.  I'm going to mark this one as answered and lock this Discussion.
